### DISCUSSION TITLE
Can the pyboard be connected to a USB camera to capture pictures

### DISCUSSION THREAD

**Discussion by Meekdai** [micropython/micropython#11835] [STM32 / Pyboard]
I want to capture images from a normal USB camera. Has anyone done this before?

**Comment by jimmo** [comment]
That would require the pyboard to be in USB Host mode (i.e. via the OTG peripheral). We only support Device mode at the moment.

**Reply by Meekdai** [reply]
Thanks @jimmo 
I found that @01studio-lab implements a similar function using ESP32-S2/S3.
https://github.com/01studio-lab/micropython/blob/master/ports/esp32/modusbcam.c
